enhancements/machine-config/machine-config-non-reconcilable-changes.md

```diff
@@ -0,0 +1,422 @@
+configuration while already existing nodes should not report updates in any 
+of the MCD supported fields. The diff between the current and the desired 
+configuration in old nodes should match all the changes applied to new nodes.
+- Scale up with the feature turned on and feature disabling: This test has is 
+divided into two sub-tests:
+    - Irreconcilable changes rolled back before validations are turned on: In 
+    this scenario `updateReports` will be cleared as the machines will match 
+    the latest rendered MC, no matter if the machine were or not added after 
+    the test introduced irreconcilable changes.
+
+    - Irreconcilable changes not rolled back before validations are turned on: 
+    In this scenario `updateReports` will continue reporting differences in 
+    "old" nodes even after disabling the feature.
+
+### Graduation Criteria
+
+This feature is behind a new FeatureGate, called NonReconcilableMC, and part
+of the tech-preview FeatureGate in 4.20. 
+Once it is tested by QE and users it can be GA'd since it should not impact 
+daily usage of a cluster.
+
+## Dev Preview -> Tech Preview
+
+Not applicable. Feature introduced in Tech Preview. 
+
+## Tech Preview -> GA
+
+- Bugs found by e2e tests and QE are solved and the dedicated e2e's are stable.
+- User facing documentation created in
+[https://github.com/openshift/openshift-docs](openshift-docs).
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+Upgrades or downgrades are not impacted by the presence or not of this feature.
+
+### Version Skew Strategy
+
+Not applicable.
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+If the irreconcilable configuration validation is performed and it fails 
+the MCO continues to report the failure as it is alraedy doing in the MCP, by
+setting to the MCP the `RenderDegraded` condition to true.
+
+If the configuration reaches the MCD and the irreconcilable validation 
+fails the MCN `UpdatePrepared` condition is updated with the details of the 
+validation failure.
+
+#### Support Procedures
+
+None.
+
+## Implementation History
+
+Not applicable.
+
+## Alternatives (Not Implemented)
+
+The main alternative to the proposal this enhancement proposes is to modify 
+the behavior of the custom MachineConfigPools to allow not inheriting from the 
+worker pool. That way, customers would be able to specify their own MCPs for 
+a dedicated set of nodes without the need of dealing with the originally 
+created worker MCP.
+While this approach can be seen as the clean way of solving the uses cases 
+this enhancement addresses is has some drawbacks that need to be mentioned:
+- It would introduce a complex breaking change in the behavior of the MCPs. A 
+custom MCP 'bootable' approach could introduce a mechanism to select whether 
+MCPs inherit from the worker pool. However, this would add unnecessary 
+complexity for customers who do not require the use cases targeted by this 
+enhancement.
```

Comment:
I'm not sure I'm following along with what the breaking change that would be made here is. Any chance we could elaborate a bit more on this to describe how this would impact existing users that don't care for this functionality?

What exactly would be breaking? There are no ways that it could be implemented to _not_ be breaking?